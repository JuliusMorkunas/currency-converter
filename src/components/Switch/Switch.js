import React from 'react';
import PropTypes from 'prop-types';
import s from './Switch.module.scss';

const Switch = ({ onClick }) => {
  return (
    <button className={s.Button} onClick={onClick}>
      <span className={s.Icon} />
    </button>
  );
};

Switch.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default Switch;
