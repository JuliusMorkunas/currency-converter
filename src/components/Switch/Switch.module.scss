@import '../../variables';

.Button {
  height: 40px;
  cursor: pointer;
  border: none;
  background-color: $c-white;
}

.Icon {
  display: inline-block;
  width: 25px;
  height: 22px;
  background-image: url('./icon-switch.svg');
  background-repeat: no-repeat;
  background-position: center;
}
