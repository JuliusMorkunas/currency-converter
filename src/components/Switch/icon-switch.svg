<svg width="27" height="24" xmlns="http://www.w3.org/2000/svg">
  <g fill="#3583EE" fill-rule="evenodd">
    <path d="M12.3789062 5.640625h11.2109375l-4.21875-4.0625c-.1822925-.16927168-.2734375-.37109258-.2734375-.60546875s.0813794-.43945225.2441407-.61523438C19.5045581.18163976 19.7031238.09375 19.9375.09375c.2343762 0 .4427074.07812422.625.234375L26.3046875 5.875c.1822926.16927168.2734375.37434775.2734375.61523438 0 .24088662-.0911449.44596269-.2734375.61523437l-5.78125 5.56640625c-.1692717.1562508-.3645822.234375-.5859375.234375-.2473971 0-.4557283-.0911449-.625-.2734375-.1692717-.1692717-.2506511-.3710926-.2441406-.6054687.0065104-.2343762.0944002-.4361971.2636718-.6054688l4.21875-4.0625H12.359375c-.2213553 0-.4166658-.07812422-.5859375-.234375-.1692717-.15625078-.2604166-.34505098-.2734375-.56640625-.0130209-.24739707.0683585-.46223867.2441406-.64453125.1757822-.18229258.3873686-.2734375.6347656-.2734375zM14.69787175 18.359375H3.4869343l4.21874995 4.0625c.18229258.1692717.2734375.3710926.2734375.6054688 0 .2343761-.08137939.4394522-.24414062.6152343-.16276124.1757822-.36132696.2636719-.59570313.2636719s-.44270742-.0781242-.625-.234375L.7720905 18.125c-.1822926-.1692717-.2734375-.3743478-.2734375-.6152344 0-.2408866.0911449-.4459627.2734375-.6152343l5.78125-5.5664063c.16927168-.1562508.36458223-.234375.5859375-.234375.24739707 0 .45572832.0911449.625.2734375.16927168.1692717.25065107.3710926.24414063.6054688-.00651045.2343761-.0944002.436197-.26367188.6054687l-4.21874995 4.0625H14.717403c.22135527 0 .41666582.0781242.5859375.234375.16927168.1562508.2604166.345051.2734375.5664063.0130209.247397-.0683585.4622386-.24414063.6445312-.17578212.1822926-.38736855.2734375-.63476562.2734375z"/>
  </g>
</svg>
