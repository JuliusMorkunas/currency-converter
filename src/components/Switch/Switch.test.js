import React from 'react';
import { render } from '@testing-library/react';
import Switch from './Switch';

it('Switch should match snapshot', () => {
  const { asFragment } = render(<Switch onClick={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});
