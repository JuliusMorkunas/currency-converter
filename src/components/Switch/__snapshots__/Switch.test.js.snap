// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="Button"
  >
    <span
      class="Icon"
    />
  </button>
</DocumentFragment>
`;
